cpp/velox/substrait/SubstraitToVeloxPlanValidator.cc

```diff
@@ -150,9 +151,18 @@ bool SubstraitToVeloxPlanValidator::validateScalarFunction(
   if (name == "extract") {
     return validateExtractExpr(params);
   }
-  if (name == "regexp_extract_all" && !scalarFunction.arguments()[1].value().has_literal()) {
-    logValidateMsg("native validation failed due to: pattern is not literal for regex_extract_all.");
-    return false;
+  if (name == "regexp_extract_all" || name == "regexp_extract" || name == "regexp_replace" || name == "rlike") {
+    auto patternArg = scalarFunction.arguments()[1].value();
+    if (!patternArg.has_literal()) {
+      logValidateMsg("native validation failed due to: pattern is not literal for regex_extract_all.");
+      return false;
+    }
+    auto pattern = patternArg.literal().string();
+    auto re2 = RE2(pattern, RE2::Quiet);
```

Comment:
Could you extract **re2** related code to utils? Maybe it will be used frequently.

Reply:
Sorry for missing this. Will do.